### PR TITLE
update quantize.py code version

### PR DESCRIPTION
**Pull request by XiaoheYou**

### CODE REVIEW

File: easycv/models/detection/utils/boxes.py [hunk 1]
```diff
@@ -76,3 +78,37 @@ def postprocess(prediction, num_classes, conf_thre=0.7, nms_thre=0.45):
             output[i] = torch.cat((output[i], detections))
 
     return output
+
+
+def yolox_predict(outputs, img_metas=None):
```
Comment (wenmengzhou) [inline easycv/models/detection/utils/boxes.py:83]: why place this function in boxes_util?

File: easycv/toolkit/quantize/quantize_utils.py [hunk 2]
```diff
@@ -1,19 +1,179 @@
...
+        quantize_config['non_traceable_module_class'] = [YOLOXHead]
+
+    return quantize_config
 
 
 def calib(model, data_loader):
     for cur_iter, data in enumerate(data_loader):
-        input_args, kwargs = scatter_kwargs(None, data, [-1])
+        if cur_iter > 50:
+            return
```
Comment (wenmengzhou) [inline easycv/toolkit/quantize/quantize_utils.py:40]: experiment code?

File: easycv/toolkit/quantize/quantize_utils.py [hunk 3]
```diff
@@ -1,19 +1,179 @@
...
+        input_args, kwargs = scatter_kwargs(None, data, [torch.cuda.current_device()]) #[-1]
+        print('calib number: {}'.format((cur_iter)))
```
Comment (wenmengzhou) [inline easycv/toolkit/quantize/quantize_utils.py:42]: this log is printed too frequently

File: easycv/toolkit/quantize/quantize_utils.py [hunk 4]
```diff
@@ -1,19 +1,179 @@
...
         with torch.no_grad():
-            kwargs[0]['img'] = kwargs[0]['img'].squeeze(dim=0)
+            # for cpu test,  use this:
+            # kwargs[0]['img'] = kwargs[0]['img'].squeeze(dim=0)
             model(kwargs[0]['img'])
```
Comment (wenmengzhou) [inline easycv/toolkit/quantize/quantize_utils.py:46]: cpu gpu test logic may use if condition to make better compatibility

File: easycv/toolkit/quantize/quantize_utils.py [hunk 5]
```diff
@@ -1,19 +1,179 @@
...
+        interpreter.runSession(session)
+
+        # 获取MNN模型的输出
+        output_tensor = interpreter.getSessionOutputAll(session)
+
+        # 获取模型输出shape
+        output_shape = model_output_shape(cfg.model.type, imgs_per_gpu)
+
+        # 将MNN模型的输出转化为PyTorch的tensor
+        tmp_output = MNN.Tensor(output_shape, MNN.Halide_Type_Float,
```
Comment (wenmengzhou) [inline easycv/toolkit/quantize/quantize_utils.py:140]: use english comment instead

File: docs/source/tutorials/compression.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
+# YOLOX Compression Tutorial
```
Comment (wenmengzhou) [inline docs/source/tutorials/compression.md:1]: quantization tutorial

File: docs/source/tutorials/compression.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+To use pai-itag detection format data to eval detection, you can refer to [configs/detection/yolox/yolox_s_8xb16_300e_coco_pai.py](../../configs/detection/yolox/yolox_s_8xb16_300e_coco_pai.py) for more configuration details.
+
+## Local & PAI-DSW
+
+To use COCO format data, use config file `configs/detection/yolox/yolox_s_8xb16_300e_coco.py`
+
+To use PAI-Itag format data, use config file `configs/detection/yolox/yolox_s_8xb16_300e_coco_pai.py`
+
+
+### Compression
```
Comment (wenmengzhou) [inline docs/source/tutorials/compression.md:40]: Qunantization

**Review by wenmengzhou — CHANGES_REQUESTED** [review]
request changes

File: easycv/apis/test.py [hunk 8]
```diff
@@ -40,7 +40,6 @@ def single_cpu_test(model,
         # use scatter_kwargs to unpack DataContainer data for raw torch.nn.module
         input_args, kwargs = scatter_kwargs(None, data, [-1])
         with torch.no_grad():
-            kwargs[0]['img'] = kwargs[0]['img'].squeeze(dim=0)
```
Comment (Cathy0908) [inline easycv/apis/test.py:43]: whether removing this line affect other models ？

File: easycv/models/detection/utils/boxes.py (hunk 1 above)
Comment (Cathy0908) [inline easycv/models/detection/utils/boxes.py:83]: this appears to be yolox post process rather than yolox predict

**Review by Cathy0908 — APPROVED** [review]

File: easycv/toolkit/quantize/quantize_utils.py [hunk 9]
```diff
@@ -1,19 +1,179 @@
...
+    if 'YOLOX' in model_type:
+        quantize_config['non_traceable_module_class'] = [YOLOXHead]
+
+    return quantize_config
 
 
 def calib(model, data_loader):
     for cur_iter, data in enumerate(data_loader):
-        input_args, kwargs = scatter_kwargs(None, data, [-1])
+        if cur_iter > 50:
```
Comment (Cathy0908) [inline easycv/toolkit/quantize/quantize_utils.py:39]: why set to 50,  what is the purpose of `calib`?

File: tools/quantize.py [hunk 10]
```diff
@@ -179,36 +190,50 @@ def main():
...
+    model.to('cpu')
+
     # quantizing model
+    print(f'convert model')
     quantized_backbone, _ = convert(prepared_backbone, quantize_config)
     model.backbone = quantized_backbone
     model.eval()
 
     # cpu eval
+    print(f'quantized model eval')
```
Comment (Cathy0908) [inline tools/quantize.py:235]: replace print with logging, other print too

**Review by Cathy0908 — APPROVED** [review]

File: easycv/toolkit/quantize/quantize_utils.py (hunk 9 above)
Comment (XiaoheYou) [reply easycv/toolkit/quantize/quantize_utils.py:39]: yes

File: easycv/apis/test.py (hunk 8 above)
Comment (XiaoheYou) [reply easycv/apis/test.py:43]: no

File: docs/source/tutorials/compression.md (hunk 7 above)
Comment (XiaoheYou) [reply docs/source/tutorials/compression.md:40]: ?

File: docs/source/tutorials/compression.md (hunk 6 above)
Comment (XiaoheYou) [reply docs/source/tutorials/compression.md:1]: this tutorial includes pruning

File: easycv/toolkit/quantize/quantize_utils.py (hunk 4 above)
Comment (XiaoheYou) [reply easycv/toolkit/quantize/quantize_utils.py:46]: If calib use cpu, it will cost too much time.

**Review by Cathy0908 — COMMENTED** [review]
revert approve

File: docs/source/tutorials/compression.md (hunk 6 above)
Comment (wenmengzhou) [reply docs/source/tutorials/compression.md:1]: Pruning and Quantization

File: docs/source/tutorials/compression.md (hunk 7 above)
Comment (wenmengzhou) [reply docs/source/tutorials/compression.md:40]: Section name is not accurate as before

File: easycv/toolkit/quantize/quantize_utils.py (hunk 9 above)
Comment (wenmengzhou) [reply easycv/toolkit/quantize/quantize_utils.py:39]: add some comment for this magic number

**Review by wenmengzhou — COMMENTED** [review]
add comment

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**